File: Logic/logic.pl
% -*- prolog -*-
% Using Swi-Prolog 
% swipl -g main -s logic.pl

message(hello, 'Hello, World.').
message(goodbye, 'Goodbye, World.').

hello(STRING) :- 
    message(hello, MESSAGE),
    STRING = MESSAGE.

main :-
    hello(MESSAGE),
    write(MESSAGE),
    write('\n'),
    halt.

File: Logic/logic.clj
;; lein2 exec logic.clj

(ns hello-world.logic
  (:refer-clojure :exclude [==])
  (:use [clojure.core.logic]))

(defrel message category message)
(fact message :hello "Hello, World.")
(fact message :bye "Goodbye, World.")

(println 
  (first
  (run* [q]
    (message :hello q))))
